src/SignalR/clients/ts/signalr/src/HandshakeProtocol.ts

```diff
@@ -40,7 +40,20 @@ export class HandshakeProtocol {
             // optional content after is additional messages
             const responseLength = separatorIndex + 1;
             messageData = String.fromCharCode.apply(null, binaryData.slice(0, responseLength));
-            remainingData = (binaryData.byteLength > responseLength) ? binaryData.slice(responseLength).buffer : null;
+
+            if (binaryData.byteLength > responseLength) {
+                remainingData = binaryData.slice(responseLength);
+                // IE 11 does not have .slice() on Uint8Array, we polyfill it with Array.prototype.slice()
+                // which returns an Array object, that doesn't have .buffer, which is an ArrayBuffer
+                // so we need to convert it into a Uint8Array and .buffer that
+                if (remainingData.buffer) {
+                    remainingData = remainingData.buffer;
+                } else {
+                    remainingData = new Uint8Array(remainingData).buffer;
+                }
+            } else {
+                remainingData = null;
+            }
```

Comment:
Does this mean we can change the signature from `(data: any): [any, HandshakeResponseMessage]` to `(data: any): [ArrayBuffer, HandshakeResponseMessage]`?

Reply:
We could probably change it to `(data: any): [ArrayBuffer | String, HandshakeResponseMessage]`